torchgeo/datasets/cropharvest.py

```diff
@@ -0,0 +1,317 @@
+    """
+
+    # *https://github.com/nasaharvest/cropharvest/blob/main/cropharvest/bands.py
+    all_bands = [
+        "VV",
+        "VH",
+        "B2",
+        "B3",
+        "B4",
+        "B5",
+        "B6",
+        "B7",
+        "B8",
+        "B8A",
+        "B9",
+        "B11",
+        "B12",
+        "temperature_2m",
+        "total_precipitation",
+        "elevation",
+        "slope",
+        "NDVI",
+    ]
+    rgb_bands = ["B11", "B8", "B2"]
+
+    features_url = "https://zenodo.org/records/7257688/files/features.tar.gz?download=1"
+    labels_url = "https://zenodo.org/records/7257688/files/labels.geojson?download=1"
+    file_dict = {
+        "features": {
+            "url": features_url,
+            "filename": "features.tar.gz",
+            "extracted_filename": os.path.join("features", "arrays"),
+            "md5": "cad4df655c75caac805a80435e46ee3e",
+        },
+        "labels": {
+            "url": labels_url,
+            "filename": "labels.geojson",
+            "extracted_filename": "labels.geojson",
+            "md5": "bf7bae6812fc7213481aff6a2e34517d",
+        },
+    }
+
+    directory = "CropHarvest"
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[dict[str, object]], dict[str, object]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new CropHarvest dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "val"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.files = self._load_features(self.root, self.directory)
+        self.labels = self._load_labels(self.root, self.directory)
+
+    def __getitem__(self, index: int) -> dict[str, object]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        data = self._load_array(files["chip"])
+
+        label = self._load_label(files["index"], files["dataset"])
+        sample = {"data": data, "label": label}
```

Comment:
We don't really have any other single-pixel datasets so not sure what to call the dict keys.